hw/formal/formal_conn.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Script to run connectivity test using Jasper Gold
+#
+# Usage: To run connectivity test on system level modules, type
+#   formal_conn -f top_earlgrey.csv
+#
+# More options:
+# -top: which top_level module to run
+# -p:  provide core file path
+# -gui: run with GUI
+# -cov: run coverage
+# -f: path to the csv file
+# -t: choose which formal tool to use, current only support jaspergold
+#
+# Example:
+#   formal_conn -gui -cov -f top_earlgrey.csv
+#
+
+export TOP="top_earlgrey"
+gui=0
+tool="jg"
+export COV=0
+export CSV_PATH
+
+while [ "$1" != "" ]; do
+  case "$1" in
+    "-top")
+      shift
+      TOP=$1
+      ;;
+    "-p")
+      shift
+      CORE_PATH=$1
+      ;;
+    "-gui")
+      gui=1
+      echo "using GUI mode"
+      ;;
+    "-cov")
+      COV=1
+      echo "enable coverage"
+      ;;
+    "-t")
+      shift
+      tool=$1
+      echo "running in tool: $tool"
+      ;;
+    "-f")
+      shift
+      CSV_PATH=../../../conn_csvs/$1
+      echo "csv path: $CSV_PATH"
+      ;;
+  esac
+  shift
+done
+
+echo "-------------------------------------------------------------------------"
+echo "-- Generate file list using FuseSoC"
+echo "-------------------------------------------------------------------------"
+echo ""
+echo "Top: $TOP"
+echo ""
+
+\rm -Rf build jgproject
+# we just run the setup for the default target in order to generate the filelist
+if [ "${CORE_PATH}" == "" ]; then
+  CORE_PATH=systems:top_earlgrey
+fi
```

Comment:
I'd consider replacing this with `CORE_PATH="${CORE_PATH:-$DEFAULT_CORE_PATH}"`, where `DEFAULT_CORE_PATH=systems:top_earlgrey` appears at the top of the script. It's simpler (no `if`), and moves the magic value "top_earlgrey" up to the top of the file.

Assuming you don't want to consume environment variables from outside, I think you probably want an `unset CORE_PATH` at the top of the script (otherwise `env CORE_PATH=foo hw/formal/formal_conn.sh` will pick up `CORE_PATH=foo`).

Reply:
Yeah that is a good point! Thanks Rupert. It is clearer here if we just set a default value and override it if user want to set a different path.